### PR TITLE
docs(fab): document FAB

### PR DESCRIPTION
**Pull request by e111077**
rendered: https://github.com/material-components/material-web/blob/fab-docs/docs/components/fab.md

### CODE REVIEW

**Comment by e111077** [comment]
Sorry this is external but ping for review

**Comment by AndrewJakubowicz** [comment]
Reviewing now - sorry for missing this!

File: docs/components/fab.md [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+
+1.  [Extended FAB with both icon and label text](#with-icon)
+2.  [Extended FAB without icon](#without-icon)
+
+## Usage
+
+### FAB
+
+FAB comes in two variants, icon-only and extended (icon with label). There are
+also three sub variants to each, surface, primary, secondary, tertiary, and
```
Comment (AndrewJakubowicz) [inline docs/components/fab.md:100]: I got confused here. There are "three sub variants", but then 5 variants seem to be listed (surface, primary, secondary, tertiary, branded FAB). Or do each of these 5 have three sub variants?

File: docs/components/fab.md [hunk 2]
```diff
@@ -0,0 +1,634 @@
...
+<!-- catalog-include "figures/fab/usage-without-icon.html" -->
+
+```html
+<md-fab label="reroute"></md-fab>
+```
+
+## Accessibility
+
+For each non-extended FAB, an `aria-label` should be provided as the icon text
+is not typcially considered accessible. If `aria-label` is not provided, the FAB
```
Comment (AndrewJakubowicz) [inline docs/components/fab.md:285]: typo: `typcially` -> `typically`

File: docs/components/fab.md [hunk 3]
```diff
@@ -0,0 +1,634 @@
...
+
+```html
+<md-fab aria-label="Edit">
+  <md-icon slot="icon">edit</md-icon>
+</md-fab>
+```
+
+### Extended FAB
+
+For each extended FAB, accessiblity is handled by the `label` attribute. By
```
Comment (AndrewJakubowicz) [inline docs/components/fab.md:296]: `accessiblity` -> `accessibility`

File: docs/components/figures/fab/extended-fabs.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<div class="figure-wrapper">
```
Comment (AndrewJakubowicz) [inline docs/components/figures/fab/extended-fabs.html:1]: It is immensely cool how these code samples are rendered into images.

File: docs/components/figures/fab/extended-fabs.html [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  <figure
+      class="types-image"
+      style="justify-content:center;"
+      title="Elevated, filled, filled tonal, outlined, and text buttons"
```
Comment (AndrewJakubowicz) [inline docs/components/figures/fab/extended-fabs.html:5]: Is this title right?

File: docs/components/figures/fab/extended-fabs.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      aria-label="The 5 types of common buttons">
```
Comment (AndrewJakubowicz) [inline docs/components/figures/fab/extended-fabs.html:6]: Update to describe fab

File: docs/components/figures/fab/fabs.html [hunk 7]
```diff
@@ -0,0 +1,65 @@
+<div class="figure-wrapper">
+  <figure
+      class="types-image"
+      style="justify-content:center;"
+      title="Elevated, filled, filled tonal, outlined, and text buttons"
+      aria-label="The 5 types of common buttons">
```
Comment (AndrewJakubowicz) [inline docs/components/figures/fab/fabs.html:6]: Please update with fab

File: docs/components/figures/fab/usage-large.html [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<div class="figure-wrapper">
+  <figure
+      aria-label="A standard surface fab"
```
Comment (AndrewJakubowicz) [inline docs/components/figures/fab/usage-large.html:3]: Nit: "A standard _large_ surface fab"

File: docs/components/figures/fab/usage-lowered.html [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<div class="figure-wrapper">
+  <figure
+      aria-label="A standard surface fab"
```
Comment (AndrewJakubowicz) [inline docs/components/figures/fab/usage-lowered.html:3]: Nit: mention this is a lowered fab?

**Review by AndrewJakubowicz — COMMENTED** [review]
Awesome!

**Comment by asyncLiz** [comment]
my nit: can we import and do the review in critique for tracking? 😅

**Comment by e111077** [comment]
will import to critique rn

**Pull request merged**